include/geode/model/helpers/component_mesh_vertices.h

```diff
@@ -0,0 +1,58 @@
+/*
+ * Copyright (c) 2019 - 2022 Geode-solutions
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ *
+ */
+
+#pragma once
+
+#include <absl/types/span.h>
+
+#include <geode/model/common.h>
+#include <geode/model/mixin/core/component_type.h>
+
+namespace geode
+{
+    struct MeshComponentVertex;
+} // namespace geode
+
+namespace geode
+{
+    struct MeshComponentVertexIntersection
+    {
+        MeshComponentVertexIntersection( ComponentID component_id_in,
+            index_t vertex0_in,
+            index_t vertex1_in )
+            : component_id{ std::move( component_id_in ) },
+              vertex0{ vertex0_in },
+              vertex1{ vertex1_in }
+        {
+        }
+
+        ComponentID component_id;
+        index_t vertex0;
+        index_t vertex1;
+    };
+
+    std::vector< MeshComponentVertexIntersection >
```

Comment:
when I read the name of the function, I thought that it returns only the MCV that are in both Spans. That does not. What does it do?

Reply:
map<ComponentID, vector< pair<index_t> >

Reply:
ComponentMeshVertexPairs component_mesh_vertex_pairs